client/src/components/Login.js

```diff
@@ -0,0 +1,66 @@
+import React, { useState } from "react";
+import "semantic-ui-css/semantic.min.css";
+import { loginUser } from "../api/auth";
+
+const Login = () => {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [error, setError] = useState(null);
+  const handleChange = event => {
+    if (event.target.name === "email") {
+      setEmail(event.target.value);
+    }
+    if (event.target.name === "password") {
+      setPassword(event.target.value);
+    }
+  };
+  const signIn = () => {
+    loginUser(email, password)
+      .then(data => {
+        const token = data.token;
+
+        localStorage.setItem("token", token);
+        localStorage.setItem("user", JSON.stringify(data.user));
+        document.location.reload();
+      })
+      .catch(() => {
+        setError(true);
+      });
+  };
+  const token = localStorage.getItem("token");
+  if (token) {
+    return <div>You are already logged in.</div>;
+  } else
+    return (
+      <div>
+        <form class="ui form">
+          <div class="field">
+            <label>Email</label>
+            <input
+              onChange={handleChange}
+              name="email"
+              type="email"
+              placeholder="Example@email.com"
+            />
+          </div>
+          <div class="field">
+            <label>Password</label>
+            <input
+              onChange={handleChange}
+              name="password"
+              type="password"
+              placeholder=" Enter password"
+            />
+          </div>
+          <button onClick={signIn} class="ui button">
+            Sign in
+          </button>
+          {error ? (
+            <div>The login details are not correct. Please try again</div>
+          ) : null}
+        </form>
```

Comment:
Nice job updating this to use a form, labels, and Semantic UI styling! However, in this project we want to use Semantic UI React components so instead we want to import them into this file and use them like so:

```
import { Button, Container, Form, Input } from 'semantic-ui-react';

<Form onSubmit={signIn}>
  <Form.Field>
    <label for="email">Email</label>
    <Input
      id="email"
      onChange={handleChange}
      name="email"
      type="email"
      placeholder="Example@email.com"
    />
  </Form.Field>
  <Form.Field>
    <label for="password">Password</label>
    <Input
      id="password"
      onChange={handleChange}
      name="password"
      type="password"
      placeholder=" Enter password"
    />
  </Form.Field>
  <Button>Sign in</Button>
  {error ? (
    <div>The login details are not correct. Please try again</div>
  ) : null}
</Form>
```

There's a few things to note:

- Using Semantic UI means we don't need to add Semantic UI class names because the components have them inside!
- Labels and inputs need to be connected using `for` and `id` (see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label for documentation on this)—this is very important for web accessibility, which is being discussed more in the frontend community. This also means that by clicking on the label (Email or Password), the input is automatically selected
- When using forms, pressing `Enter` when in an input or clicking on a button will submit the form. Therefore, we need to move the `signIn` function to the form `onSubmit` event.

Hope that all makes sense!